File: app/src/main/java/com/glucocontrol/domain/usecase/auth/SignUpWithEmailUseCase.kt
package com.glucocontrol.domain.usecase.auth

import com.glucocontrol.domain.repository.AuthRepository
import javax.inject.Inject

class SignUpWithEmailUseCase @Inject constructor(private val authRepository: AuthRepository) {
    suspend operator fun invoke(email: String, password: String) = authRepository.signUpWithEmail(email, password)
}

File: lib/domain/usecase/auth/sign_up_with_email_use_case.dart
import 'package:glucocontrol/domain/repository/auth_repository.dart';

class SignUpWithEmailUseCase {
  final AuthRepository _repo;
  SignUpWithEmailUseCase(this._repo);
  Future<void> call(String email, String password) =>
      _repo.signUpWithEmail(email, password);
}
